### PR TITLE
3.0: Check if PCluster image is already present before calling the create

### PR DESCRIPTION
**Pull request by lukeseawalker**
* Check if image with same name is already present before calling the create
* Placeholder for the logic to handle bucket resources and upload of the artifacts

**Please See** [Git Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions)

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/models/imagebuilder.py [hunk 1]
```diff
@@ -144,39 +146,61 @@ def config(self):
...
+            )
 
-        validation_failures = self.config.validate()
-        if validation_failures:
-            # TODO skip validation errors
-            raise ImageBuilderActionError("Configuration is invalid", validation_failures=validation_failures)
+        if not suppress_validators:
+            validation_failures = self.config.validate()
+            for failure in validation_failures:
+                if failure.level.value > FailureLevel(validation_failure_level).value:
```
Comment (yuleiwan) [inline cli/src/pcluster/models/imagebuilder.py:172]: should it be >=?

File: cli/src/pcluster/models/imagebuilder.py [hunk 2]
```diff
@@ -144,39 +146,61 @@ def config(self):
...
+                    # Raise the exception if there is a failure with a level greater than the specified one
+                    raise ImageBuilderActionError("Configuration is invalid", validation_failures=validation_failures)
```
Comment (yuleiwan) [inline cli/src/pcluster/models/imagebuilder.py:174]: `validation_failures=validation_failures`, should we give all the validation failures or just throw failure level greater than or equal to given min failure level?

File: cli/src/pcluster/models/imagebuilder.py [hunk 3]
```diff
@@ -144,39 +146,61 @@ def config(self):
...
 
         # Add tags information to the stack
-        tags = copy.deepcopy(self.config.build.tags) or []
-        tags.append(BaseTag(key="pcluster_build_image", value=get_installed_version()))
-        tags = [{"Key": tag.key, "Value": tag.value} for tag in tags]
+        cfn_tags = copy.deepcopy(self.config.build.tags) or []
+        cfn_tags.append(BaseTag(key="pcluster_build_image", value=get_installed_version()))
+        cfn_tags = [{"Key": tag.key, "Value": tag.value} for tag in cfn_tags]
```
Comment (yuleiwan) [inline cli/src/pcluster/models/imagebuilder.py:179]: we can add a TODO to reminder adding cloudwatch tag and s3 bucket tag in cfn_tags.

Comment (lukeseawalker) [reply cli/src/pcluster/models/imagebuilder.py:179]: done

File: cli/src/pcluster/models/imagebuilder.py (hunk 1 above)
Comment (lukeseawalker) [reply cli/src/pcluster/models/imagebuilder.py:172]: fixed

File: cli/src/pcluster/constants.py [hunk 4]
```diff
@@ -56,3 +56,5 @@
 CW_LOGS_RETENTION_DAYS_DEFAULT = 14
 CW_DASHBOARD_ENABLED_DEFAULT = True
 CW_LOGS_ENABLED_DEFAULT = True
+
+PCLUSTER_IMAGE_NAME_TAG = "parallelcluster:image_name"
```
[the remark this answers is not in the document]
Comment (lukeseawalker) [reply cli/src/pcluster/constants.py:60]: not sure I get what you mean

File: cli/src/pcluster/models/imagebuilder.py (hunk 2 above)
Comment (lukeseawalker) [reply cli/src/pcluster/models/imagebuilder.py:174]: we can choose to return all or the subset

File: cli/src/pcluster/constants.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (yuleiwan) [reply cli/src/pcluster/constants.py:60]: I mean we should align the tag format, e.g. `pcluster_* -> parallelcluster:*` and update the validator.

**Review by yuleiwan — APPROVED** [review]

File: cli/src/pcluster/constants.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (lukeseawalker) [reply cli/src/pcluster/constants.py:60]: oh yes, this in on the todo list

**Pull request merged**